x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/support/xcontent/XContentSource.java

```diff
@@ -109,6 +109,17 @@ public <T> T getValue(String path) {
         return (T) ObjectPath.eval(path, data());
     }
 
+    /**
+     * Extracts a value identified by the given path in the source.
+     *
+     * @param path the path segments to the requested value
+     * @return The extracted value or {@code null} if no value is associated with the given path
+     */
+    @SuppressWarnings("unchecked")
+    public <T> T getValue(String[] path) {
```

Comment:
The reason why this method was needed shows that we are probably making life hard for downstream consumers - that's us when we write the telemetry-side code - by having field names that begin with dots.

We could instead strip the leading dots from the next layer of field names. That would then avoid the need for this new method, and the test could demonstrate that it's possible to navigate to the most deeply nested values using the dots-as-fieldname-separators notation.

Reply:
The background to this is that Elasticsearch interprets `.` in the field name as nested objects, `{foo.bar: 42}` is actually indexed as `{foo: {bar: 42}}`.

Reply:
OK, I've removed the leading dot from the internal model IDs.

I'm not thrilled about slightly modifying the IDs, but I agree that the leading dots are inconvenient in other code. E.g. Removing them already saves me 18 lines of code right away.